### PR TITLE
Coverage as in https://github.com/google/oss-fuzz/issues/26

### PR DESCRIPTION
**Pull request by vitalybuka**

### CODE REVIEW

File: docs/new_library.md [hunk 1]
```diff
@@ -202,6 +202,12 @@ $ python scripts/helper.py run_fuzzer $LIB_NAME name_of_a_fuzzer
 If everything works locally, then it should also work on our automated builders
 and ClusterFuzz.
 
+It's recommended look at coverage as a sanity check to make sure that fuzzer gets to the code you expect.
```
Comment (mikea) [inline docs/new_library.md]: recommended to

File: scripts/helper.py [hunk 2]
```diff
@@ -230,6 +233,62 @@ def run_fuzzer(run_args):
...
+  temp_dir = tempfile.mkdtemp()
+
+  command = [
+      'docker', 'run', '-i',
+      '-v', '%s:/out' % os.path.join(BUILD_DIR, 'out'),
+      '-v', '%s:/cov' % temp_dir,
+      '-w', '/cov',
+      '-e', 'ASAN_OPTIONS=coverage=1,detect_leaks=0',
+      '-t', 'ossfuzz/libfuzzer-runner',
+      '/out/%s/%s' % (args.library_name, args.fuzzer_name),
```
Comment (mikea) [inline scripts/helper.py]: I'm surprised by this additional directory. AFAIK our build scripts do not create second level.

File: scripts/helper.py [hunk 3]
```diff
@@ -230,6 +233,62 @@ def run_fuzzer(run_args):
...
+      '-max_total_time=%s' % args.run_time
+  ] + args.fuzzer_args
+
+  print('Running:', _get_command_string(command))
+  pipe = subprocess.Popen(command)
+  pipe.communicate()
+
+  checkout_dir = os.path.join(BUILD_DIR, args.library_name)
```
Comment (mikea) [inline scripts/helper.py]: you don't have to run it in docker. You can run all of it on host => no need to install in the image python.

File: scripts/helper.py (hunk 2 above)
Comment (vitalybuka) [inline scripts/helper.py]: Not sure what do you mean.

File: scripts/helper.py (hunk 3 above)
Comment (vitalybuka) [inline scripts/helper.py]: maybe it's insecure to run random python script downloaded over http

Comment (mikea) [inline scripts/helper.py]: Hm...

Can you try running it within standard image? For example, one of these: https://hub.docker.com/_/python/

Comment (mikea) [inline scripts/helper.py]: If not, let's create a special image for server. I am not keen on pulling new packages just for this.

File: docs/new_library.md [hunk 4]
```diff
@@ -202,6 +202,12 @@ $ python scripts/helper.py run_fuzzer $LIB_NAME name_of_a_fuzzer
 If everything works locally, then it should also work on our automated builders
 and ClusterFuzz.
 
+It's recommended to look at coverage as a sanity check to make sure that fuzzer gets to the code you expect.
+
+```bash
+$ sudo python scripts/helper.py run_with_coverage $LIB_NAME name_of_a_fuzzer seconds_to_run
```
Comment (vitalybuka) [inline docs/new_library.md]: done

File: infra/base-images/all.sh [hunk 5]
```diff
@@ -19,4 +19,4 @@ docker build --pull -t ossfuzz/base $@ infra/base-images/base
 docker build -t ossfuzz/base-clang $@ infra/base-images/base-clang
 docker build -t ossfuzz/base-libfuzzer $@ infra/base-images/base-libfuzzer
 docker build -t ossfuzz/libfuzzer-runner $@ infra/base-images/libfuzzer-runner
-
+docker build -t ossfuzz/coverage $@ infra/base-images/coverage
```
Comment (mikea) [inline infra/base-images/all.sh]: You will also need to change Jenkinsfile in this folder & Jenkinsfile in ../push-images/

File: scripts/helper.py [hunk 6]
```diff
@@ -230,6 +233,61 @@ def run_fuzzer(run_args):
   pipe = subprocess.Popen(command)
   pipe.communicate()
 
+def coverage(run_args):
+  """Runs a fuzzer in the container."""
+  parser = argparse.ArgumentParser('helper.py coverage')
+  parser.add_argument('library_name', help='name of the library')
+  parser.add_argument('fuzzer_name', help='name of the fuzzer')
+  parser.add_argument('run_time', help='time in seconds to run fuzzer')
```
Comment (mikea) [inline scripts/helper.py]: let's add default (60s?)

Comment (vitalybuka) [inline scripts/helper.py]: done

File: infra/base-images/all.sh (hunk 5 above)
Comment (vitalybuka) [inline infra/base-images/all.sh]: done

**Pull request merged**